scrapy_poet/utils.py

```diff
@@ -121,27 +84,31 @@ def is_response_going_to_be_used(request, spider):
     if is_callback_using_response(callback):
         return True
 
-    plan, _ = build_plan(callback, None)
-    for provider in get_providers(plan):
+    for provider in get_providers(callback):
         if is_provider_using_response(provider):
             return True
 
     return False
 
 
-def get_providers(plan: andi.Plan):
-    for obj, _ in plan:
-        provider = providers.get(obj)  # type: ignore
-        if not provider:
-            continue
+def get_providers(callback):
+    for argument, possible_types in andi.inspect(callback).items():
+        for cls in possible_types:
+            if is_injectable(cls):
+                yield from get_providers(cls)
+                continue
```

Comment:
Semantics..
`get_providers` and `providers.get`
Not sure if the distinction is clear.
Maybe import `providers` as `available_providers` and `get_providers` to `search_for_providers`, as it is not `get`, but most like a search in a tree

Reply:
Maybe `get_callback_providers` or `get_providers_from_callback`.

Reply:
Oh, I see your point. Yes, it's more like a search indeed.

Reply:
I've just renamed it to `discover_callback_providers`. What do you think?